src/services/RoomService.ts

```diff
@@ -0,0 +1,184 @@
+        const newRoom: Types.Room = {
+            roomId: roomId,
+            owner: ownerId,
+            requestSent: false,
+            invited: invited,
+        };
+
+        newRoom.invited[ownerId].accepted = true;
+        this.rooms[roomId] = newRoom;
+        logger.logInfo(`${ownerId} has created room with id ${roomId}`);
+        return roomId;
+    }
+
+    /**
+     * Connects user to room
+     *
+     * @param userId - user id connecting to room
+     * @param roomId - room id user is connecting to
+     */
+    connectToRoom(userId: string, roomId: string) {
+        const room = this.getRoom(roomId);
+        room.invited[userId].accepted = true;
+        logger.logInfo(`${userId} connected to room ${roomId}`);
+        let invitedUserIds = Object.keys(room.invited);
+        invitedUserIds.forEach(socketId => {
+            if (room.invited[socketId].accepted) {
+                io.to(socketId).emit(Constants.ROOM_STATUS, {
+                    type: Constants.USER_CONNECT,
+                    roomId: roomId,
+                    full: invitedUserIds.every(invitedUserId => {
+                        return room.invited[invitedUserId].accepted;
+                    }),
+                    owner: room.owner,
+                    userId: userId,
+                });
+            }
+        });
+    }
+
+    /**
+     * Deletes room
+     *
+     * @param roomId - room id
+     */
+    deleteRoom(roomId: string) {
+        delete this.rooms[roomId];
+    }
+
+    /**
+     * Handles when a user leaves a room
+     *
+     * @param roomId - room id
+     * @param userId - user id
+     */
+    handleLeaveRoom(roomId: string, userId: string) {
+        if (this.rooms[roomId]) {
+            const room = this.rooms[roomId];
+            // Remove from room
+            delete room.invited[userId];
+            logger.logInfo(`${userId} disconnected from room ${roomId}`);
+    
+            // Update users in current room or delete room if empty.
+            let invitedUserIds = Object.keys(room.invited);
+            if (invitedUserIds.length === 1) {
+                io.to(invitedUserIds[0]).emit(Constants.LEAVE_ROOM, { roomId: roomId });
+                this.deleteRoom(roomId);
+                logger.logInfo(`Everyone left room ${roomId} and it has been closed.`);
+            } else {
+                invitedUserIds.forEach(socketId => {
+                    if (room.invited[socketId].accepted) {
+                        io.to(socketId).emit(Constants.ROOM_STATUS, {
+                            type: Constants.USER_DISCONNECT,
+                            roomId: roomId,
+                            invited: room.invited,
+                            full: invitedUserIds.every(invitedUserId => {
+                                return room.invited[invitedUserId].accepted;
+                            }),
+                            owner: room.owner,
+                            userId: userId,
```

Comment:
i think you can can use object type inference here to assign `roomId` and `userId` like:
```
{
  type: Constants.USER_DISCONNECT,
  roomId,
  invited: room.invited,
  full: invitedUserIds.every(invitedUserId => {
    return room.invited[invitedUserId].accepted;
  }),
  owner: room.owner,
  userId,
}
```

Reply:
I know, but the problem if we do it this way is if we need to replace the variable names (like we did for roomid -> roomId), it gets confusing if we don't know the original key name in the object.

Reply:
Hm, ok i guess it's fine for now, but maybe we can add types for each of the socket message types later?

Reply:
Yea that would work.